editor/src/core/shared/github.ts

```diff
@@ -173,15 +173,30 @@ export async function saveProjectToGithub(
   projectID: string,
   persistentModel: PersistentModel,
   dispatch: EditorDispatch,
+  branchName: string | null,
+  commitMessage: string | null,
```

Comment:
This typing makes me very uneasy, because it leaves us susceptible to accidentally messing up the order when calling the function. I think we should replace it with some form of (probably nullable) `options` object.